make/modules/java.base/lib/CoreLibraries.gmk

```diff
@@ -53,19 +53,22 @@ endif
 $(eval $(call SetupJdkLibrary, BUILD_LIBJAVA, \
     NAME := java, \
     OPTIMIZATION := HIGH, \
+    EXTRA_HEADER_DIRS := \
+        libjava/nio/ch, \
     jdk_util.c_CFLAGS := $(VERSION_CFLAGS), \
     ProcessImpl_md.c_CFLAGS := $(VERSION_CFLAGS), \
     java_props_md.c_CFLAGS := \
         -DARCHPROPNAME='"$(OPENJDK_TARGET_CPU_OSARCH)"', \
     DISABLED_WARNINGS_gcc_ProcessImpl_md.c := unused-result, \
     JDK_LIBS := libjvm, \
-    LIBS_linux := $(LIBDL), \
+    LIBS_linux := $(LIBDL) -lpthread, \
     LIBS_aix := $(LIBDL) $(LIBM),\
     LIBS_macosx := \
         -framework CoreFoundation \
+        -framework CoreServices \
         -framework Foundation \
         -framework SystemConfiguration, \
-    LIBS_windows := advapi32.lib ole32.lib shell32.lib version.lib, \
+    LIBS_windows := advapi32.lib mswsock.lib ole32.lib shell32.lib version.lib ws2_32.lib, \
```

Comment:
Are these libraries still required to build libnio, or can they be removed from that library now?

Reply:
I will check. Thanks for noticing it.

Reply:
`CoreServices` is still necessary.

Reply:
The Windows libs are also still required for `TransmitFile` and `WSAGetLastError`:

- ```mswsock.lib: FileDispatcherImpl.obj : error LNK2019: unresolved external symbol TransmitFile```
- ```ws2_32.lib: FileDispatcherImpl.obj : error LNK2019: unresolved external symbol __imp_WSAGetLastError```

Reply:
And pthread?

Reply:
I will check.

Reply:
`pthread` is still needed:
```
open/src/java.base/unix/native/libjava/nio/ch/NativeThread.c:83: error: undefined reference to 'pthread_kill'
````

Reply:
Thank you for suggesting it!